lib/nfc/helpers/iso14443_4_layer.c

```diff
@@ -54,11 +65,39 @@ bool iso14443_4_layer_decode_block(
 
     bool ret = false;
 
-    do {
-        if(!bit_buffer_starts_with_byte(block_data, instance->pcb_prev)) break;
-        bit_buffer_copy_right(output_data, block_data, 1);
-        ret = true;
-    } while(false);
+    if(iso14443_4_layer_is_block_chaining(instance, block_data)) {
+        do {
+            bit_buffer_copy_right(output_data, block_data, 1);
+            ret = true;
+        } while(false);
+    } else {
+        do {
+            if(!bit_buffer_starts_with_byte(block_data, instance->pcb_prev)) break;
+            size_t used = bit_buffer_get_size_bytes(output_data);
+            size_t capacity = bit_buffer_get_capacity_bytes(output_data);
+            size_t available = capacity - used;
+            size_t needed = bit_buffer_get_size_bytes(block_data);
+            if(needed > available) break;
+            FURI_LOG_D(
+                TAG,
+                "used=%zu, capacity=%zu, available=%zu, needed=%zu",
+                used,
+                capacity,
+                available,
+                needed);
+
+            bit_buffer_append_right(output_data, block_data, 1);
```

Comment:
This is where it fails if output_data is non-empty (even when output_data has more than enough room for block_data)